src/gwc/src/main/java/org/geoserver/gwc/wms/CachingWebMapService.java

```diff
@@ -64,7 +66,14 @@ public WebMap invoke(MethodInvocation invocation) throws Throwable {
 
         final GetMapRequest request = getRequest(invocation);
         boolean tiled = request.isTiled() || !config.isRequireTiledParameter();
-        if (!tiled) {
+        final Map<String, String> rawKvp = request.getRawKvp();
+        boolean isSeedingRequest =
+                rawKvp != null
+                        && rawKvp.containsKey(
+                                GeoServerTileLayer
+                                        .GWC_SEED_INTERCEPT_TOKEN); // skipping seeding requests to
+        // avoid meta tile recursion
```

Comment:
Making peace with the formatter:

```suggestion
        // skipping seeding requests to avoid meta tile recursion
        boolean isSeedingRequest =
                rawKvp != null && rawKvp.containsKey(GeoServerTileLayer.GWC_SEED_INTERCEPT_TOKEN);
```